code/__HELPERS/mobs.dm

```diff
@@ -192,8 +192,14 @@ GLOBAL_LIST_EMPTY(species_list)
 
 	var/target_loc = target.loc
 
-	LAZYADD(user.do_afters, target)
-	LAZYADD(target.targeted_by, user)
+	if(!source && target)
+		source = target //Use the direct ref to the target
```

Comment:
the var name of source is likely to be confusing, and i look forward to 2 years from now hearing somebody in coderbus bitching about how the name doesn't make sense for what it does when they dare to enter this proc.

Reply:
yeah i wanted to change it, I was going for something similar to traits.